parseen254/LinkTreeClone
# Feat: Firebase Deployment and Github Actions

## Description
This pull request adds the necessary configuration files for deploying the project to Firebase Hosting. The changes include:

- Add `.firebaserc` file to specify the default Firebase project
- Add `firebase.json` file to configure Firebase Hosting settings
- Add GitHub Actions workflows for automatic deployment to Firebase Hosting on merge to the main branch and on pull requests

## Changes
- `.firebaserc`: This file specifies the default Firebase project ID as "link-tree-clone-c7e0a".
- `firebase.json`: This file configures the public directory for deployment as ".svelte-kit/output" and specifies files and directories to ignore during deployment.
- `.github/workflows/firebase-hosting-merge.yml`: This GitHub Actions workflow automatically deploys the project to Firebase Hosting when changes are merged into the main branch.
- `.github/workflows/firebase-hosting-pull-request.yml`: This GitHub Actions workflow deploys a preview of the project to Firebase Hosting for each pull request, allowing for easy review and testing.

## Benefits
- Automated deployment to Firebase Hosting, reducing manual effort and ensuring consistent deployments.
- Preview deployments for pull requests, enabling easier collaboration and review.
- Clear separation of deployment configuration from the main codebase.

## Testing
- Ensure that the Firebase project ID specified in `.firebaserc` is correct.
- Verify that the public directory specified in `firebase.json` matches the build output directory of your project.
- Create a pull request to trigger the preview deployment workflow and check that the preview URL is accessible.
- Merge the pull request into the main branch and confirm that the automatic deployment to Firebase Hosting is successful.
